docs/pktoken.md

```diff
@@ -0,0 +1,702 @@
+      "typ": "COS"
+    },
+    "signature":"hiyDh..."
+  }
+]}
+```
+
+#### Nonce-Commitment
+
+Our example PK Token uses a *nonce-commitment* to bind the identity's public key `upk` (user public key) to the ID Token. That is, the `nonce` value in the payload commits to CIC's protected header which includes the identity's public key. By commits we mean that the nonce has been set to be the hash of the CIC's protected header:
+
+```ascii
+nonce = SHA3(
+  Base64({"alg":"ES256",   
+    "rz":"301a510ffd19b4888cdec7b9dda62192cfa06d85936cabb1afdd1a015ad44137",
+    "typ":"CIC",
+    "upk":{"alg":"ES256","crv":"P-256","kty":"EC" "x":"8JAMvpmdrhiKJi9A79LHPj5CPKlyztHHEkCr6tntyq8", "y":"jRqKcX8wIU24ffb5GI6z9XlePqlP1DOxvlEwvp0wC5s"}
+  }))
+```
+
+Notice that the value which is hashed to generate the `nonce` value in the payload is exactly the value given in the protected header of the CIC. Put another way, the CIC allows the verifier to open the commitment to the identity's public key in the nonce. The value `rz` is randomly chosen by the identity's OpenPubkey client, a.k.a. the client-instance, to ensure that each time a `nonce` is generated, it is always unique and random. The value `upk` is the identity's public key.
+
+#### MFA Cosigner Signature
+
+We have included in a Cosigner signature in this example to show what what one looks like. Given that the Cosigner signature does not not differ much between examples we have omitted it from the other examples. Note that PK Tokens can function without a Cosigner signature, a verifier can choose to require or not require one based on a verifiers security needs.
+
+### GQ Signed Nonce-Commitment PK Tokens - Google Example
+
+This PK Token is for the same Google account as the prior example including the same use of the nonce-commitment, the only difference is that we have replaced Google's RSA signature and protected header with a GQ Signature.
+
+```JSON
+{
+  "payload": {
+    "iss": "https://accounts.google.com",
+    "azp": "992028499768-ce9juclb3vvckh23r83fjkmvf1lvjq18.apps.googleusercontent.com",
+    "aud": "992028499768-ce9juclb3vvckh23r83fjkmvf1lvjq18.apps.googleusercontent.com",
+    "sub": "104852002444754136271",
+    "email": "anon.author.aardvark@gmail.com",
+    "email_verified": true,
+    "at_hash": "7y8qnEw17J6BDoqvz6ydbg",
+    "nonce": "8IpXCsOcYBGcCJmXJMFOpBjz4-kPXwDhYi3hm_DFM_U",
+    "name": "Anonymous Author",
+    "picture": "https://lh3.googleusercontent.com/a/ACg8ocIdbWtaAGFsizjWVh7Q6C-XDBuSoUOpf7d7nGqgNQ-9yHmenNA=s96-c",
+    "given_name": "Anonymous",
+    "family_name": "Author",
+    "iat": 1722717291,
+    "exp": 1722720891
+  },
+  "signatures": [
+    {
+      "protected": {
+      "alg": "GQ256",
+      "jkt": "w0bhEOa9d4qxGtKLGhwySJ2VZtRxPA-0abeIC9C-zPQ",
+      "kid": "eyJhbGciOiJSUzI1NiIsImtpZCI6ImUyNmQ5MTdiMWZlOGRlMTMzODJhYTdjYzlhMWQ2ZTkzMjYyZjMzZTIiLCJ0eXAiOiJKV1QifQ",
+      "typ": "JWT"
+      },
+      "signature": "hAo4...(5504 base64 characters)"
```

Comment:
Is this signature really 5000 chars long?

Reply:
GQ signatures are huge. They are basically a blinded RSA signature repeated under different challenge values 256  times.